File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/cadastro/perfil/PerfilGridDTO.java
package br.com.grupopipa.gestaointegrada.cadastro.perfil;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class PerfilGridDTO implements GridDTO {
    private UUID id;
    private String nome;
    private LocalDateTime createdAt;
}

File: src/frontend/src/app/components/cadastro/perfil/model/perfil-grid-dto.ts
export class PerfilGridDTO {
  id: string;
  nome: string;
  createdAt: Date;
  deleted?: boolean;

  constructor(id: string, nome: string, createdAt: Date, deleted?: boolean) {
    this.id = id;
    this.nome = nome;
    this.createdAt = createdAt;
    this.deleted = deleted;
  }
}
